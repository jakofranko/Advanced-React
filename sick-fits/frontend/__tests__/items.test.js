import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponet from '../components/Item';

const fakeItem = {
    id: 'foo',
    title: 'bar',
    price: 1000,
    description: 'baz',
    image: 'unittest.jpg',
    largeImage: 'largeTest.jpg'
};

describe('<Item />', () => {

    it('should render snapshot properly', () => {
        const comp = shallow(<ItemComponet item={fakeItem} />);
        expect(toJSON(comp)).toMatchSnapshot();
    })
    it('should render properly', () => {
        const comp = shallow(<ItemComponet item={fakeItem} />);
        const priceTag = comp.find('PriceTag');
        expect(priceTag.dive().text()).toBe('$10');
        expect(comp.find('Title a').text()).toBe(fakeItem.title);
    });

    it('renders the image properly', () => {
        const comp = shallow(<ItemComponet item={fakeItem} />);
        expect(comp.find('img').props().src).toBe(fakeItem.image);
        expect(comp.find('img').props().alt).toBe(fakeItem.title);
    });

    it('renders the buttons properly', () => {
        const comp = shallow(<ItemComponet item={fakeItem} />);
        const buttonList = comp.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        expect(buttonList.find('Link')).toHaveLength(1);
        expect(buttonList.find('AddToCart').exists()).toBe(true);
        expect(buttonList.find('DeleteItem').exists()).toBe(true);
    });
});
