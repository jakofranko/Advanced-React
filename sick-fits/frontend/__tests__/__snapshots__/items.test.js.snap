// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render snapshot properly 1`] = `
<ItemStyles__Item>
  <img
    alt="bar"
    src="unittest.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "foo",
          },
        }
      }
    >
      <a>
        bar
      </a>
    </Link>
  </Title>
  <PriceTag>
    $10
  </PriceTag>
  <p>
    baz
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "foo",
          },
        }
      }
    >
      <a>
        Edit ‚úè
      </a>
    </Link>
    <AddToCart
      id="foo"
    />
    <DeleteItem
      id="foo"
    />
  </div>
</ItemStyles__Item>
`;
